File: Services/ShippingService/ShippingService.BLL/Services/ShipmentService.cs
using AutoMapper;
using ShippingService.BLL.Models;
using ShippingService.BLL.Services.Interfaces;
using ShippingService.DAL.Entities;
using ShippingService.DAL.Repositories.Interfaces;

namespace ShippingService.BLL.Services
{
    public class ShipmentService : IShipmentService
    {
        private readonly IShipmentRepository _shipmentRepository;
        private readonly IMapper _mapper;

        public ShipmentService(IShipmentRepository customerRepository, IMapper mapper)
        {
            _shipmentRepository = customerRepository;
            _mapper = mapper;
        }
        public async Task<Shipment> Create(Shipment model)
        {
            var entity = _mapper.Map<ShipmentEntity>(model);
            var result = await _shipmentRepository.Create(entity);

            return _mapper.Map<Shipment>(result);
        }

        public async Task<Shipment?> GetByOrderId(int orderId)
        {
            var entity = await _shipmentRepository.GetByOrderId(orderId);

            return _mapper.Map<Shipment>(entity);
        }
    }
}

File: Client/ivcrm-react-client/src/services/ShipmentService.ts
import { IChangeShipment } from "../models/IChangeShipment";
import { IShipment } from "../models/IShipment";
import axiosInstance from "./AxiosInstance";

export default class ShipmentService {
    static async create(shipment: IChangeShipment) {
        const response = await axiosInstance.post<IShipment>(`/shipment`, shipment)
        return response.data;
    }

    static async getByOrderId(orderId: number) {
        const response = await axiosInstance.get<Array<IShipment>>(`/shipment/order/${orderId}`)
        return response.data;
    }
}
 